Mathlib/CategoryTheory/Limits/Shapes/FunctorToTypes.lean

```diff
@@ -0,0 +1,261 @@
+/-
+Copyright (c) 2024 Jack McKoen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jack McKoen
+-/
+import Mathlib.CategoryTheory.Limits.FunctorCategory
+import Mathlib.CategoryTheory.Limits.Types
+
+/-!
+# Binary (co)products of type-valued functors
+
+Defines an explicit construction of binary products and coproducts of type-valued functors.
+
+Also defines isomorphisms to the categorical product and coproduct, respectively.
+-/
+
+
+open CategoryTheory.Limits
+
+universe w v u
+
+namespace CategoryTheory.FunctorToTypes
+
+variable {C : Type u} [Category.{v} C]
+
+section prod
+
+/-- `prod F G` is the explicit binary product of type-valued functors `F` and `G`. -/
+def prod (F G : C ⥤ Type w) : C ⥤ Type w where
+  obj a := F.obj a × G.obj a
+  map f a := (F.map f a.1, G.map f a.2)
+
+  /-- The first projection of `prod F G`, onto `F`. -/
+@[simps]
+def prod.fst {F G : C ⥤ Type w} : (prod F G) ⟶ F where
+  app _ a := a.1
+
+/-- The second projection of `prod F G`, onto `G`. -/
+@[simps]
+def prod.snd {F G : C ⥤ Type w} : (prod F G) ⟶ G where
+  app _ a := a.2
+
+/-- Given natural transformations `F ⟶ F₁` and `F ⟶ F₂`, construct
+a natural transformation `F ⟶ prod F₁ F₂`. -/
+def natTransProd {F F₁ F₂ : C ⥤ Type w} (τ₁ : F ⟶ F₁) (τ₂ : F ⟶ F₂) :
```

Comment:
In order to mimic the product API naming, I would suggest this name:
```suggestion
def prod.lift {F F₁ F₂ : C ⥤ Type w} (τ₁ : F ⟶ F₁) (τ₂ : F ⟶ F₂) :
```
It would be nice to add equational (simp) lemmas like `prod.lift_fst : prod.lift τ₁ τ₂ ≫prod.fst = τ₁`. (The `apply` version of these should be obtained by adding the `@[simps]` attribute to `prod.lift`.